{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyPRb2cxKkDJKdrSuzBd+pho"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"0nPQO1c6jrn1","executionInfo":{"status":"ok","timestamp":1738045095011,"user_tz":420,"elapsed":2699,"user":{"displayName":"Janani Aiswarya Ravichandran","userId":"07746560599602780693"}},"outputId":"5221dcce-d060-4258-f4a3-fda3f8475de5"},"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: google-auth in /usr/local/lib/python3.11/dist-packages (2.27.0)\n","Requirement already satisfied: google-auth-oauthlib in /usr/local/lib/python3.11/dist-packages (1.2.1)\n","Requirement already satisfied: google-auth-httplib2 in /usr/local/lib/python3.11/dist-packages (0.2.0)\n","Requirement already satisfied: google-api-python-client in /usr/local/lib/python3.11/dist-packages (2.155.0)\n","Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from google-auth) (5.5.1)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from google-auth) (0.4.1)\n","Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.11/dist-packages (from google-auth) (4.9)\n","Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from google-auth-oauthlib) (1.3.1)\n","Requirement already satisfied: httplib2>=0.19.0 in /usr/local/lib/python3.11/dist-packages (from google-auth-httplib2) (0.22.0)\n","Requirement already satisfied: google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5 in /usr/local/lib/python3.11/dist-packages (from google-api-python-client) (2.19.2)\n","Requirement already satisfied: uritemplate<5,>=3.0.1 in /usr/local/lib/python3.11/dist-packages (from google-api-python-client) (4.1.1)\n","Requirement already satisfied: googleapis-common-protos<2.0.dev0,>=1.56.2 in /usr/local/lib/python3.11/dist-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (1.66.0)\n","Requirement already satisfied: protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<6.0.0.dev0,>=3.19.5 in /usr/local/lib/python3.11/dist-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (4.25.5)\n","Requirement already satisfied: proto-plus<2.0.0dev,>=1.22.3 in /usr/local/lib/python3.11/dist-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (1.25.0)\n","Requirement already satisfied: requests<3.0.0.dev0,>=2.18.0 in /usr/local/lib/python3.11/dist-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (2.32.3)\n","Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /usr/local/lib/python3.11/dist-packages (from httplib2>=0.19.0->google-auth-httplib2) (3.2.1)\n","Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /usr/local/lib/python3.11/dist-packages (from pyasn1-modules>=0.2.1->google-auth) (0.6.1)\n","Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.11/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib) (3.2.2)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (3.4.1)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (3.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (2.3.0)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests<3.0.0.dev0,>=2.18.0->google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client) (2024.12.14)\n"]}],"source":["!pip install google-auth google-auth-oauthlib google-auth-httplib2 google-api-python-client\n"]},{"cell_type":"code","source":["!pip install google-api-python-client==1.7.2\n","!pip install google-auth==1.8.0\n","!pip install google-auth-httplib2==0.0.3\n","!pip install google-auth-oauthlib==0.4.1\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":1000},"id":"y09ej7SYoPIN","outputId":"b0fc0edc-e146-47c6-b251-97c0270eb4a4"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting google-api-python-client==1.7.2\n","  Downloading google_api_python_client-1.7.2-py2.py3-none-any.whl.metadata (1.2 kB)\n","Requirement already satisfied: httplib2<1dev,>=0.9.2 in /usr/local/lib/python3.11/dist-packages (from google-api-python-client==1.7.2) (0.22.0)\n","Requirement already satisfied: google-auth>=1.4.1 in /usr/local/lib/python3.11/dist-packages (from google-api-python-client==1.7.2) (2.27.0)\n","Requirement already satisfied: google-auth-httplib2>=0.0.3 in /usr/local/lib/python3.11/dist-packages (from google-api-python-client==1.7.2) (0.2.0)\n","Requirement already satisfied: six<2dev,>=1.6.1 in /usr/local/lib/python3.11/dist-packages (from google-api-python-client==1.7.2) (1.17.0)\n","Collecting uritemplate<4dev,>=3.0.0 (from google-api-python-client==1.7.2)\n","  Downloading uritemplate-3.0.1-py2.py3-none-any.whl.metadata (4.6 kB)\n","Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from google-auth>=1.4.1->google-api-python-client==1.7.2) (5.5.1)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from google-auth>=1.4.1->google-api-python-client==1.7.2) (0.4.1)\n","Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.11/dist-packages (from google-auth>=1.4.1->google-api-python-client==1.7.2) (4.9)\n","Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /usr/local/lib/python3.11/dist-packages (from httplib2<1dev,>=0.9.2->google-api-python-client==1.7.2) (3.2.1)\n","Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /usr/local/lib/python3.11/dist-packages (from pyasn1-modules>=0.2.1->google-auth>=1.4.1->google-api-python-client==1.7.2) (0.6.1)\n","Downloading google_api_python_client-1.7.2-py2.py3-none-any.whl (54 kB)\n","\u001b[?25l   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/54.9 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m54.9/54.9 kB\u001b[0m \u001b[31m1.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading uritemplate-3.0.1-py2.py3-none-any.whl (15 kB)\n","Installing collected packages: uritemplate, google-api-python-client\n","  Attempting uninstall: uritemplate\n","    Found existing installation: uritemplate 4.1.1\n","    Uninstalling uritemplate-4.1.1:\n","      Successfully uninstalled uritemplate-4.1.1\n","  Attempting uninstall: google-api-python-client\n","    Found existing installation: google-api-python-client 2.155.0\n","    Uninstalling google-api-python-client-2.155.0:\n","      Successfully uninstalled google-api-python-client-2.155.0\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","earthengine-api 1.4.6 requires google-api-python-client>=1.12.1, but you have google-api-python-client 1.7.2 which is incompatible.\n","pydrive2 1.21.3 requires google-api-python-client>=1.12.5, but you have google-api-python-client 1.7.2 which is incompatible.\n","firebase-admin 6.6.0 requires google-api-python-client>=1.7.8, but you have google-api-python-client 1.7.2 which is incompatible.\u001b[0m\u001b[31m\n","\u001b[0mSuccessfully installed google-api-python-client-1.7.2 uritemplate-3.0.1\n"]},{"output_type":"display_data","data":{"application/vnd.colab-display-data+json":{"pip_warning":{"packages":["googleapiclient","uritemplate"]},"id":"6c7245a7f9e7455894fccdfa37c897a1"}},"metadata":{}},{"output_type":"stream","name":"stdout","text":["Collecting google-auth==1.8.0\n","  Downloading google_auth-1.8.0-py2.py3-none-any.whl.metadata (3.3 kB)\n","Collecting cachetools<3.2,>=2.0.0 (from google-auth==1.8.0)\n","  Downloading cachetools-3.1.1-py2.py3-none-any.whl.metadata (4.6 kB)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from google-auth==1.8.0) (0.4.1)\n","Collecting rsa<4.1,>=3.1.4 (from google-auth==1.8.0)\n","  Downloading rsa-4.0-py2.py3-none-any.whl.metadata (3.0 kB)\n","Requirement already satisfied: setuptools>=40.3.0 in /usr/local/lib/python3.11/dist-packages (from google-auth==1.8.0) (75.1.0)\n","Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.11/dist-packages (from google-auth==1.8.0) (1.17.0)\n","Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /usr/local/lib/python3.11/dist-packages (from pyasn1-modules>=0.2.1->google-auth==1.8.0) (0.6.1)\n","Downloading google_auth-1.8.0-py2.py3-none-any.whl (75 kB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m75.8/75.8 kB\u001b[0m \u001b[31m2.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hDownloading cachetools-3.1.1-py2.py3-none-any.whl (11 kB)\n","Downloading rsa-4.0-py2.py3-none-any.whl (38 kB)\n","Installing collected packages: cachetools, rsa, google-auth\n","  Attempting uninstall: cachetools\n","    Found existing installation: cachetools 5.5.1\n","    Uninstalling cachetools-5.5.1:\n","      Successfully uninstalled cachetools-5.5.1\n","  Attempting uninstall: rsa\n","    Found existing installation: rsa 4.9\n","    Uninstalling rsa-4.9:\n","      Successfully uninstalled rsa-4.9\n","  Attempting uninstall: google-auth\n","    Found existing installation: google-auth 2.27.0\n","    Uninstalling google-auth-2.27.0:\n","      Successfully uninstalled google-auth-2.27.0\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","google-colab 1.0.0 requires google-auth==2.27.0, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-functions 1.19.0 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-generativeai 0.8.4 requires google-auth>=2.15.0, but you have google-auth 1.8.0 which is incompatible.\n","pandas-gbq 0.26.1 requires google-auth>=2.13.0, but you have google-auth 1.8.0 which is incompatible.\n","gspread 6.1.4 requires google-auth>=1.12.0, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-iam 2.17.0 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-bigtable 2.28.1 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","bigframes 1.33.0 requires google-auth<3.0dev,>=2.15.0, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-firestore 2.19.0 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-storage 2.19.0 requires google-auth<3.0dev,>=2.26.1, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-translate 3.19.0 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-bigquery-connection 1.17.0 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","pydata-google-auth 1.9.0 requires google-auth<3.0dev,>=1.25.0, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-bigquery 3.25.0 requires google-auth<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-datastore 2.20.2 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-auth-oauthlib 1.2.1 requires google-auth>=2.15.0, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-aiplatform 1.74.0 requires google-auth<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","earthengine-api 1.4.6 requires google-api-python-client>=1.12.1, but you have google-api-python-client 1.7.2 which is incompatible.\n","google-cloud-resource-manager 1.14.0 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","pydrive2 1.21.3 requires google-api-python-client>=1.12.5, but you have google-api-python-client 1.7.2 which is incompatible.\n","google-cloud-pubsub 2.25.0 requires google-auth<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-ai-generativelanguage 0.6.15 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-bigquery-storage 2.27.0 requires google-auth<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","firebase-admin 6.6.0 requires google-api-python-client>=1.7.8, but you have google-api-python-client 1.7.2 which is incompatible.\n","google-cloud-language 2.16.0 requires google-auth!=2.24.0,!=2.25.0,<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-genai 0.3.0 requires google-auth<3.0.0dev,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","pymc 5.19.1 requires cachetools>=4.2.1, but you have cachetools 3.1.1 which is incompatible.\n","google-api-core 2.19.2 requires google-auth<3.0.dev0,>=2.14.1, but you have google-auth 1.8.0 which is incompatible.\n","google-cloud-core 2.4.1 requires google-auth<3.0dev,>=1.25.0, but you have google-auth 1.8.0 which is incompatible.\u001b[0m\u001b[31m\n","\u001b[0mSuccessfully installed cachetools-3.1.1 google-auth-1.8.0 rsa-4.0\n"]},{"output_type":"display_data","data":{"application/vnd.colab-display-data+json":{"pip_warning":{"packages":["cachetools","google","rsa"]},"id":"207c21b749f548238f849e59fd3c013e"}},"metadata":{}},{"output_type":"stream","name":"stdout","text":["Collecting google-auth-httplib2==0.0.3\n","  Downloading google_auth_httplib2-0.0.3-py2.py3-none-any.whl.metadata (2.1 kB)\n","Requirement already satisfied: google-auth in /usr/local/lib/python3.11/dist-packages (from google-auth-httplib2==0.0.3) (1.8.0)\n","Requirement already satisfied: httplib2>=0.9.1 in /usr/local/lib/python3.11/dist-packages (from google-auth-httplib2==0.0.3) (0.22.0)\n","Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /usr/local/lib/python3.11/dist-packages (from httplib2>=0.9.1->google-auth-httplib2==0.0.3) (3.2.1)\n","Requirement already satisfied: cachetools<3.2,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-httplib2==0.0.3) (3.1.1)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-httplib2==0.0.3) (0.4.1)\n","Requirement already satisfied: rsa<4.1,>=3.1.4 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-httplib2==0.0.3) (4.0)\n","Requirement already satisfied: setuptools>=40.3.0 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-httplib2==0.0.3) (75.1.0)\n","Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-httplib2==0.0.3) (1.17.0)\n","Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /usr/local/lib/python3.11/dist-packages (from pyasn1-modules>=0.2.1->google-auth->google-auth-httplib2==0.0.3) (0.6.1)\n","Downloading google_auth_httplib2-0.0.3-py2.py3-none-any.whl (6.3 kB)\n","\u001b[31mERROR: Operation cancelled by user\u001b[0m\u001b[31m\n","\u001b[0mTraceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/cli/base_command.py\", line 179, in exc_logging_wrapper\n","    status = run_func(*args)\n","             ^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/cli/req_command.py\", line 67, in wrapper\n","    return func(self, options, args)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/commands/install.py\", line 447, in run\n","    conflicts = self._determine_conflicts(to_install)\n","                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/commands/install.py\", line 578, in _determine_conflicts\n","    return check_install_conflicts(to_install)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/operations/check.py\", line 101, in check_install_conflicts\n","    package_set, _ = create_package_set_from_installed()\n","                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/operations/check.py\", line 39, in create_package_set_from_installed\n","    for dist in env.iter_installed_distributions(local_only=False, skip=()):\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/metadata/base.py\", line 664, in <genexpr>\n","    return (d for d in it if d.canonical_name not in skip)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/metadata/base.py\", line 619, in iter_all_distributions\n","    dist.canonical_name,\n","    ^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/metadata/importlib/_dists.py\", line 170, in canonical_name\n","    name = self._get_dist_name_from_location() or get_dist_name(self._dist)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/metadata/importlib/_dists.py\", line 166, in _get_dist_name_from_location\n","    return stem.split(\"-\", 1)[0]\n","           ^^^^^^^^^^^^^^^^^^\n","KeyboardInterrupt\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/bin/pip3\", line 8, in <module>\n","    sys.exit(main())\n","             ^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/cli/main.py\", line 80, in main\n","    return command.main(cmd_args)\n","           ^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/cli/base_command.py\", line 100, in main\n","    return self._main(args)\n","           ^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/cli/base_command.py\", line 232, in _main\n","    return run(options, args)\n","           ^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/pip/_internal/cli/base_command.py\", line 215, in exc_logging_wrapper\n","    logger.critical(\"Operation cancelled by user\")\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 1536, in critical\n","    self._log(CRITICAL, msg, args, **kwargs)\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 1634, in _log\n","    self.handle(record)\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 1644, in handle\n","    self.callHandlers(record)\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 1706, in callHandlers\n","    hdlr.handle(record)\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 978, in handle\n","    self.emit(record)\n","  File \"/usr/lib/python3.11/logging/handlers.py\", line 75, in emit\n","    logging.FileHandler.emit(self, record)\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 1230, in emit\n","    StreamHandler.emit(self, record)\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 1114, in emit\n","    self.flush()\n","  File \"/usr/lib/python3.11/logging/__init__.py\", line 1094, in flush\n","    self.stream.flush()\n","KeyboardInterrupt\n","^C\n","Collecting google-auth-oauthlib==0.4.1\n","  Downloading google_auth_oauthlib-0.4.1-py2.py3-none-any.whl.metadata (2.4 kB)\n","Requirement already satisfied: google-auth in /usr/local/lib/python3.11/dist-packages (from google-auth-oauthlib==0.4.1) (1.8.0)\n","Requirement already satisfied: requests-oauthlib>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from google-auth-oauthlib==0.4.1) (1.3.1)\n","Requirement already satisfied: oauthlib>=3.0.0 in /usr/local/lib/python3.11/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib==0.4.1) (3.2.2)\n","Requirement already satisfied: requests>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib==0.4.1) (2.32.3)\n","Requirement already satisfied: cachetools<3.2,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-oauthlib==0.4.1) (3.1.1)\n","Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-oauthlib==0.4.1) (0.4.1)\n","Requirement already satisfied: rsa<4.1,>=3.1.4 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-oauthlib==0.4.1) (4.0)\n","Requirement already satisfied: setuptools>=40.3.0 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-oauthlib==0.4.1) (75.1.0)\n","Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.11/dist-packages (from google-auth->google-auth-oauthlib==0.4.1) (1.17.0)\n","Requirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /usr/local/lib/python3.11/dist-packages (from pyasn1-modules>=0.2.1->google-auth->google-auth-oauthlib==0.4.1) (0.6.1)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib==0.4.1) (3.4.1)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib==0.4.1) (3.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib==0.4.1) (2.3.0)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests>=2.0.0->requests-oauthlib>=0.7.0->google-auth-oauthlib==0.4.1) (2024.12.14)\n","Downloading google_auth_oauthlib-0.4.1-py2.py3-none-any.whl (18 kB)\n"]}]},{"cell_type":"code","source":["from google.colab import drive\n","drive.mount('/content/drive', force_remount=True)\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"mPkAQWocrYaY","executionInfo":{"status":"ok","timestamp":1738047209888,"user_tz":420,"elapsed":2268,"user":{"displayName":"Janani Aiswarya Ravichandran","userId":"07746560599602780693"}},"outputId":"0bd02c31-da4f-4aa9-fc62-d79fe70101a7"},"execution_count":13,"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}]},{"cell_type":"code","source":["from google.colab import auth\n","from google.auth.transport.requests import Request\n","from googleapiclient.discovery import build\n","from google_auth_oauthlib.flow import InstalledAppFlow\n","import os\n","\n","# Authentication\n","auth.authenticate_user()\n","\n","# Define SCOPES\n","SCOPES = ['https://www.googleapis.com/auth/documents']\n","\n","# Load credentials and create a Google Docs API service\n","creds = None\n","if os.path.exists('token.json'):\n","    creds = Credentials.from_authorized_user_file('token.json', SCOPES)\n","if not creds or not creds.valid:\n","    if creds and creds.expired and creds.refresh_token:\n","        creds.refresh(Request())\n","    else:\n","        flow = InstalledAppFlow.from_client_secrets_file('credentials.json', SCOPES)\n","        creds = flow.run_local_server(port=33965)\n","    with open('token.json', 'w') as token:\n","        token.write(creds.to_json())\n","\n","# Create the Google Docs API service\n","service = build('docs', 'v1', credentials=creds)\n","\n","# Provided markdown content\n","markdown_content = \"\"\"\n","# Product Team Sync - May 15, 2023\n","\n","## Attendees\n","- Sarah Chen (Product Lead)\n","- Mike Johnson (Engineering)\n","- Anna Smith (Design)\n","- David Park (QA)\n","\n","## Agenda\n","\n","### 1. Sprint Review\n","* Completed Features\n","  * User authentication flow\n","  * Dashboard redesign\n","  * Performance optimization\n","    * Reduced load time by 40%\n","    * Implemented caching solution\n","* Pending Items\n","  * Mobile responsive fixes\n","  * Beta testing feedback integration\n","\n","### 2. Current Challenges\n","* Resource constraints in QA team\n","* Third-party API integration delays\n","* User feedback on new UI\n","  * Navigation confusion\n","  * Color contrast issues\n","\n","### 3. Next Sprint Planning\n","* Priority Features\n","  * Payment gateway integration\n","  * User profile enhancement\n","  * Analytics dashboard\n","* Technical Debt\n","  * Code refactoring\n","  * Documentation updates\n","\n","## Action Items\n","- [ ] @sarah: Finalize Q3 roadmap by Friday\n","- [ ] @mike: Schedule technical review for payment integration\n","- [ ] @anna: Share updated design system documentation\n","- [ ] @david: Prepare QA resource allocation proposal\n","\n","## Next Steps\n","* Schedule individual team reviews\n","* Update sprint board\n","* Share meeting summary with stakeholders\n","\n","## Notes\n","* Next sync scheduled for May 22, 2023\n","* Platform demo for stakeholders on May 25\n","* Remember to update JIRA tickets\n","\n","---\n","Meeting recorded by: Sarah Chen\n","Duration: 45 minutes\n","\"\"\"\n","\n","def create_google_doc_from_markdown(markdown_content):\n","    # Create a new Google Doc\n","    doc = service.documents().create().execute()\n","    document_id = doc['documentId']\n","\n","    # Initialize list to hold the requests for document formatting\n","    requests = []\n","\n","    # Split the markdown content and process line by line\n","    lines = markdown_content.splitlines()\n","    current_index = 1  # Start at position 1 in the document\n","\n","    for line in lines:\n","        if line.startswith('# '):  # Heading 1\n","            requests.append({\n","                'insertText': {'location': {'index': current_index}, 'text': line[2:] + '\\n'}\n","            })\n","            requests.append({\n","                'updateParagraphStyle': {\n","                    'range': {'startIndex': current_index, 'endIndex': current_index + len(line[2:]) + 1},\n","                    'paragraphStyle': {'namedStyleType': 'HEADING_1'}\n","                }\n","            })\n","            current_index += len(line) + 1\n","\n","        elif line.startswith('## '):  # Heading 2\n","            requests.append({\n","                'insertText': {'location': {'index': current_index}, 'text': line[3:] + '\\n'}\n","            })\n","            requests.append({\n","                'updateParagraphStyle': {\n","                    'range': {'startIndex': current_index, 'endIndex': current_index + len(line[3:]) + 1},\n","                    'paragraphStyle': {'namedStyleType': 'HEADING_2'}\n","                }\n","            })\n","            current_index += len(line) + 1\n","\n","        elif line.startswith('### '):  # Heading 3\n","            requests.append({\n","                'insertText': {'location': {'index': current_index}, 'text': line[4:] + '\\n'}\n","            })\n","            requests.append({\n","                'updateParagraphStyle': {\n","                    'range': {'startIndex': current_index, 'endIndex': current_index + len(line[4:]) + 1},\n","                    'paragraphStyle': {'namedStyleType': 'HEADING_3'}\n","                }\n","            })\n","            current_index += len(line) + 1\n","\n","        elif line.startswith('- [ ]'):  # Checkbox\n","            requests.append({\n","                'insertText': {'location': {'index': current_index}, 'text': line + '\\n'}\n","            })\n","            current_index += len(line) + 1\n","\n","        elif line.startswith('* '):  # Bullets\n","            requests.append({\n","                'insertText': {'location': {'index': current_index}, 'text': line[2:] + '\\n'}\n","            })\n","            current_index += len(line) + 1\n","\n","        else:  # Regular text\n","            requests.append({\n","                'insertText': {'location': {'index': current_index}, 'text': line + '\\n'}\n","            })\n","            current_index += len(line) + 1\n","\n","    # Send the requests to Google Docs API\n","    service.documents().batchUpdate(documentId=document_id, body={'requests': requests}).execute()\n","\n","    return document_id\n","\n","# Create the Google Doc from markdown content\n","doc_id = create_google_doc_from_markdown(markdown_content)\n","\n","print(f\"Document created at: https://docs.google.com/document/d/{doc_id}/edit\")\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":402},"id":"R9gOixp6tdQw","executionInfo":{"status":"error","timestamp":1738047943256,"user_tz":420,"elapsed":21069,"user":{"displayName":"Janani Aiswarya Ravichandran","userId":"07746560599602780693"}},"outputId":"4b816ce6-d3a0-4682-9b64-6c08a774679f"},"execution_count":29,"outputs":[{"output_type":"stream","name":"stdout","text":["Please visit this URL to authorize this application: https://accounts.google.com/o/oauth2/auth?response_type=code&client_id=429426087425-l9f8tfahed3gediorvg2sfcr1clmg1ue.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Flocalhost%3A33965%2F&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocuments&state=AA9YDGPfIEG8MsVFjLQQFKrwhrchgb&access_type=offline\n"]},{"output_type":"error","ename":"KeyboardInterrupt","evalue":"","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)","\u001b[0;32m<ipython-input-29-1d7acdf709a1>\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     20\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m         \u001b[0mflow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mInstalledAppFlow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrom_client_secrets_file\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'credentials.json'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mSCOPES\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m         \u001b[0mcreds\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mflow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_local_server\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mport\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m33965\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'token.json'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mtoken\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m         \u001b[0mtoken\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcreds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.11/dist-packages/google_auth_oauthlib/flow.py\u001b[0m in \u001b[0;36mrun_local_server\u001b[0;34m(self, host, port, authorization_prompt_message, success_message, open_browser, **kwargs)\u001b[0m\n\u001b[1;32m    452\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mauthorization_prompt_message\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mauth_url\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    453\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 454\u001b[0;31m         \u001b[0mlocal_server\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandle_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    455\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    456\u001b[0m         \u001b[0;31m# Note: using https here because oauthlib is very picky that\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/lib/python3.11/socketserver.py\u001b[0m in \u001b[0;36mhandle_request\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    293\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    294\u001b[0m             \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 295\u001b[0;31m                 \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mselector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mselect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    296\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mready\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    297\u001b[0m                     \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_handle_request_noblock\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/lib/python3.11/selectors.py\u001b[0m in \u001b[0;36mselect\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    413\u001b[0m         \u001b[0mready\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    414\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 415\u001b[0;31m             \u001b[0mfd_event_list\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_selector\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpoll\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    416\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mInterruptedError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    417\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mready\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mKeyboardInterrupt\u001b[0m: "]}]}]}